File: genium/src/test/resources/smoke/instances/output/cpp/src/smoke/StructWithInstances.cpp
// -------------------------------------------------------------------------------------------------
//
//
// -------------------------------------------------------------------------------------------------
#include "smoke/StructWithInstances.h"

namespace smoke {
StructWithInstances::StructWithInstances( )
    : class_instance{ }, interface_instance{ }
{
}
StructWithInstances::StructWithInstances( const ::std::shared_ptr< ::smoke::SimpleClass >& class_instance, const ::std::shared_ptr< ::smoke::SimpleInterface >& interface_instance )
    : class_instance( class_instance ), interface_instance( interface_instance )
{
}
}

File: gluecodium/src/test/resources/smoke/instances/output/android/com/example/smoke/StructWithInstances.java
/*
 *

 */
package com.example.smoke;
import android.support.annotation.NonNull;
public final class StructWithInstances {
    @NonNull
    public SimpleClass classInstance;
    @NonNull
    public SimpleInterface interfaceInstance;
    public StructWithInstances(@NonNull final SimpleClass classInstance, @NonNull final SimpleInterface interfaceInstance) {
        this.classInstance = classInstance;
        this.interfaceInstance = interfaceInstance;
    }
    @NonNull
    public native SimpleClass useSimpleClass(@NonNull final SimpleClass input);
    @NonNull
    public native SimpleInterface useSimpleInterface(@NonNull final SimpleInterface input);
}
